llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2888,25 +2885,19 @@ static const SCEV *getStrideFromPointer(Value *Ptr, ScalarEvolution *SE, Loop *L
     return nullptr;
 
   V = S->getStepRecurrence(*SE);
-  if (!V)
-    return nullptr;
 
   // Strip off the size of access multiplication if we are still analyzing the
   // pointer.
   if (OrigPtr == Ptr) {
-    if (const SCEVMulExpr *M = dyn_cast<SCEVMulExpr>(V)) {
-      if (M->getOperand(0)->getSCEVType() != scConstant)
+    if (auto *M = dyn_cast<SCEVMulExpr>(V)) {
+      auto *StepConst = dyn_cast<SCEVConstant>(M->getOperand(0));
+      if (!StepConst)
         return nullptr;
 
-      const APInt &APStepVal = cast<SCEVConstant>(M->getOperand(0))->getAPInt();
-
-      // Huge step value - give up.
-      if (APStepVal.getBitWidth() > 64)
+      if (auto StepVal = StepConst->getAPInt().trySExtValue();
```

Comment:
I personally find it more readable without multi-line if statements, i.e. something like this:

```
  auto StepVal = StepConst->getAPInt().trySExtValue();
  if (!StepVal)
```

Reply:
Agreed, I also think it makes things harder to read and it is quite uncommon across the llvm codebase AFAIK.